tests/components/airly/test_sensor.py

```diff
@@ -37,171 +23,22 @@ async def test_sensor(
-
-    state = hass.states.get("sensor.home_ozone")
-    assert state
-    assert state.state == "41.52"
-    assert state.attributes.get(ATTR_ATTRIBUTION) == ATTRIBUTION
-    assert (
-        state.attributes.get(ATTR_UNIT_OF_MEASUREMENT)
-        == CONCENTRATION_MICROGRAMS_PER_CUBIC_METER
-    )
-    assert state.attributes.get(ATTR_DEVICE_CLASS) == SensorDeviceClass.OZONE
-    assert state.attributes.get(ATTR_STATE_CLASS) == SensorStateClass.MEASUREMENT
-
-    entry = entity_registry.async_get("sensor.home_ozone")
-    assert entry
-    assert entry.unique_id == "123-456-o3"
-    assert entry.options["sensor"] == {"suggested_display_precision": 0}
-
-    state = hass.states.get("sensor.home_sulphur_dioxide")
-    assert state
-    assert state.state == "13.97"
-    assert state.attributes.get(ATTR_ATTRIBUTION) == ATTRIBUTION
-    assert (
-        state.attributes.get(ATTR_UNIT_OF_MEASUREMENT)
-        == CONCENTRATION_MICROGRAMS_PER_CUBIC_METER
-    )
-    assert state.attributes.get(ATTR_DEVICE_CLASS) == SensorDeviceClass.SULPHUR_DIOXIDE
-    assert state.attributes.get(ATTR_STATE_CLASS) == SensorStateClass.MEASUREMENT
-
-    entry = entity_registry.async_get("sensor.home_sulphur_dioxide")
-    assert entry
-    assert entry.unique_id == "123-456-so2"
-    assert entry.options["sensor"] == {"suggested_display_precision": 0}
-
-    state = hass.states.get("sensor.home_pressure")
-    assert state
-    assert state.state == "1019.86"
-    assert state.attributes.get(ATTR_ATTRIBUTION) == ATTRIBUTION
-    assert state.attributes.get(ATTR_UNIT_OF_MEASUREMENT) == UnitOfPressure.HPA
-    assert state.attributes.get(ATTR_DEVICE_CLASS) == SensorDeviceClass.PRESSURE
-    assert state.attributes.get(ATTR_STATE_CLASS) == SensorStateClass.MEASUREMENT
-
-    entry = entity_registry.async_get("sensor.home_pressure")
-    assert entry
-    assert entry.unique_id == "123-456-pressure"
-    assert entry.options["sensor"] == {"suggested_display_precision": 0}
-
-    state = hass.states.get("sensor.home_temperature")
-    assert state
-    assert state.state == "14.37"
-    assert state.attributes.get(ATTR_ATTRIBUTION) == ATTRIBUTION
-    assert state.attributes.get(ATTR_UNIT_OF_MEASUREMENT) == UnitOfTemperature.CELSIUS
-    assert state.attributes.get(ATTR_DEVICE_CLASS) == SensorDeviceClass.TEMPERATURE
-    assert state.attributes.get(ATTR_STATE_CLASS) == SensorStateClass.MEASUREMENT
+    entity_entries = er.async_entries_for_config_entry(entity_registry, entry.entry_id)
 
-    entry = entity_registry.async_get("sensor.home_temperature")
-    assert entry
-    assert entry.unique_id == "123-456-temperature"
-    assert entry.options["sensor"] == {"suggested_display_precision": 1}
+    assert entity_entries
+    for entity_entry in entity_entries:
+        assert entity_entry == snapshot(name=f"{entity_entry.entity_id}-entry")
+        assert hass.states.get(entity_entry.entity_id) == snapshot(
+            name=f"{entity_entry.entity_id}-state"
+        )
```

Comment:
```suggestion
        assert (state := hass.states.get(entity_entry.entity_id))
        assert state == snapshot(name=f"{entity_entry.entity_id}-state")
```

We make sure the state is not None (and prevent accidentally saving a None state in the snapshot)